src/main.rs

```diff
@@ -235,22 +250,21 @@ fn run(config: Config) -> Result<(), Error> {
 
     if some {
         /* We created some services, let's make sure the module is loaded */
-        let modules_load_path = "/run/modules-load.d/zram.conf";
-        let modules_load_path = Path::new(&modules_load_path);
+        let modules_load_path = Path::new(&config.root[..]).join("run/modules-load.d/zram.conf");
         let parent_path = modules_load_path.parent()
             .ok_or_else(|| format_err!("Couldn't get parent of {}", modules_load_path.display()))?;
         let _ = fs::create_dir_all(parent_path)?;
-        let mut modules_load = fs::File::create(modules_load_path).with_path(modules_load_path)?;
+        let mut modules_load = fs::File::create(&modules_load_path).with_path(&modules_load_path)?;
         modules_load.write(b"zram\n")?;
     }
 
     Ok(())
 }
 
-fn get_total_memory() -> Result<u64, Error> {
-    let path = Path::new("/proc/meminfo");
+fn get_total_memory(root: &str) -> Result<u64, Error> {
+    let path = Path::new(root).join("proc/meminfo");
```

Comment:
Here too, there is no /proc/meminfo underneath the temporary root. So the original path needs to be used.